sql-expressions.md

```diff
@@ -0,0 +1,508 @@
+or `<expr2>` otherwise.
+
+### `CASE` expressions
+
+Syntax:
+
+	  CASE <cond>
+		 WHEN <condval1> THEN <expr1>
+	   [ WHEN <condvalx> THEN <exprx> ] ...
+	   [ ELSE <expr2> ]
+	  END
+
+Evaluates `<cond>`, then picks the `WHEN` branch where `<condval>` is
+equal to `<cond>` then evaluates and returns the corresponding `THEN`
+expression. If no `WHEN` branch matches, the `ELSE` expression is
+evaluated and returned, if any, otherwise `NULL` is returned.
+
+### `NULLIF` expressions
+
+Syntax:
+
+	NULLIF(<expr1>, <expr2>)
+
+Equivalent to: `IF(<expr1> = <expr2>, NULL, <expr1>)`
+
+### `COALESCE` and `IFNULL` expressions
+
+Syntax:
+
+	IFNULL(<expr1>, <expr2>)
+	COALESCE(<expr1> [, <expr2> [, <expr3> ] ...])
+
+`COALESCE` evaluates the first expression first. If its value is not
+`NULL`, its value is returned directly. Otherwise, it returns the
+result of applying `COALESCE` on the remaining expressions. If all the
+expressions are `NULL`, `NULL` is returned.
+
+`IFNULL(a, b)` is equivalent to `COALESCE(a, b)`.
+
+### `AND` and `OR`: boolean short-circuit comparisons
+
+Syntax:
+
+	<expr1> AND <expr2>
+	<expr1> OR <expr2>
+
+
+These operators compute the boolean AND or OR function of their
+operands, using short-circuit evaluation:
+
+- Both operators first evaluate their left operand.
+- If the resulting value is `TRUE`, `OR` returns `TRUE` directly and
+  does not evaluate its right operand (short circuit), whereas `AND`
+  evaluates and returns the value of the right operand.
+- If the resulting value of the left operand is `FALSE` or `NULL`, `AND`
+  returns `FALSE` directly and does not evaluate its right operand,
+  whereas `OR` evaluates and returns the value of the right operand.
+
+## Aggregate expressions
+
+An aggregate expression has the same syntax as a function call, with a special
+case for `COUNT`:
+
+	name '(' arguments... ')'
+	COUNT '(' '*' ')'
+
+The difference between aggregate expressions and function calls is
+that the former use an
+[aggregate functions](functions-and-operators.html#aggregate-functions)
+and can only appear in the list of rendered expressions in a
+[`SELECT` clause](select.html).
+
+An aggregate expressions computes a combined value, depending on
+which aggregate function is used, across all the rows currently
+selected.
+
+## Window function calls
+
+A window function call has the syntax of a function call followed by an `OVER` clause:
+
+	name '(' arguments... ')' OVER <window>
+	name '(' '*' ')' OVER <window>
+
+It represents the application of a window or aggregate function over a
+subset ("window") of the rows selected by a query.
+
+## Explicit type coercions
+
+Syntax:
+
+	 <expr> :: <type>
+	 CAST (<expr> AS <type>)
+
+Evaluates the expression, then converts the resulting value to the
+specified type. An error is reported if the conversion is invalid.
+
+For example: `CAST(now() AS DATE)`
+
+Note that in many cases a type annotation is preferrable to a type
```

Comment:
I agree a link is useful. However with the link the user can simply follow the link to find the explanation.